app/services/foreman_templates/template_exporter.rb

```diff
@@ -96,18 +94,24 @@ def get_dump_dir(template)
     end
 
     def templates_to_dump
-      base = find_templates
-      if filter.present?
-        method = negate ? :reject : :select
-        base.public_send(method) { |template| template.name.match(/#{filter}/i) }
-      else
-        base
+      find_templates.each do |template|
+        if filter.present?
+          matches = template.name.match(/#{filter}/i)
+          exportable = matches ? true && !negate : false || negate
```

Comment:
If I changed it to `matches ? !negate : negate` would I get the same result?

Reply:
Yep, this is the same, thanks.